src/typing/flow_js.ml

```diff
@@ -3644,6 +3644,15 @@ let rec __flow cx ((l: Type.t), (u: Type.use_t)) trace =
       in
       rec_flow cx trace (o, u)
 
+    | (NumT (reason_x, Literal (n, _)), ElemT(reason_op, (ObjT _ as o), t, rw)) ->
+      let x = string_of_float_trunc n in
```

Comment:
Related comment: https://github.com/facebook/flow/issues/380#issuecomment-111329261

> wait until we have a credible replica of toString before going forward on this.

 Spec wording vaguely indicates that final result could be implementation-specific, so I'm not sure that it's worth it.

`string_of_float_trunc` seems to work good enough. Another option is to only allow integer literals as property keys.